rest-services/client/src/main/java/com/_4point/aem/docservices/rest_services/client/generatePDF/RestServicesGeneratePDFServiceAdapter.java

```diff
@@ -0,0 +1,227 @@
+import com._4point.aem.fluentforms.api.generatePDF.CreatePDFResult;
+import com._4point.aem.fluentforms.api.generatePDF.GeneratePDFService.GeneratePDFServiceException;
+import com._4point.aem.fluentforms.impl.SimpleDocumentFactoryImpl;
+import com._4point.aem.fluentforms.impl.generatePDF.CreatePDFResultImpl;
+import com._4point.aem.fluentforms.impl.generatePDF.PDFSettings;
+import com._4point.aem.fluentforms.impl.generatePDF.SecuritySettings;
+import com._4point.aem.fluentforms.impl.generatePDF.TraditionalGeneratePDFService;
+
+public class RestServicesGeneratePDFServiceAdapter extends RestServicesServiceAdapter
+implements TraditionalGeneratePDFService {
+	private static final String GENERATE_PDF_PATH = "/services/GeneratePDFService/CreatePDF";
+	private static final String DATA_PARAM_NAME = "data";
+	private static final String FILE_EXTENSION = "fileExtension";
+	private static final String FILE_TYPE_SETTINGS = "fileTypeSettings";
+	private static final String PDF_SETTINGS = "pdfSettings";
+	private static final String SECURITY_SETTINGS = "securitySettings";
+	private static final String SETTING_DOC = "settingDoc";
+	private static final String XMP_DOC = "xmpDoc";
+
+	protected RestServicesGeneratePDFServiceAdapter(WebTarget baseTarget) {
+		super(baseTarget);
+	}
+
+	// Only callable from Builder
+	private RestServicesGeneratePDFServiceAdapter(WebTarget target, Supplier<String> correlationId) {
+		super(target, correlationId);
+	}
+
+	@Override
+	public CreatePDFResult createPDF2(Document inputDoc, String inputFileExtension, String fileTypeSettings,
+			PDFSettings pdfSettings, SecuritySettings securitySettings, Document settingsDoc, Document xmpDoc)
+					throws GeneratePDFServiceException {
+		WebTarget geneWebTarget = baseTarget.path(GENERATE_PDF_PATH);
+
+		try (final FormDataMultiPart multipart = new FormDataMultiPart()) {
+			if (inputDoc != null) {
+				multipart.field(DATA_PARAM_NAME, inputDoc.getInputStream(), MediaType.MULTIPART_FORM_DATA_TYPE);
+			} else {
+				throw new NullPointerException("inputDoc can not be null");
+			}
+
+			if (inputFileExtension != null) {
+				multipart.field(FILE_EXTENSION, inputFileExtension);
+			} else {
+				throw new NullPointerException("fileextension can not be null");
+			}
+
+			MultipartTransformer.create(multipart)
+			.transform((t) -> fileTypeSettings == null ? t : t.field(FILE_TYPE_SETTINGS, fileTypeSettings))
+			.transform((t) -> pdfSettings == null ? t : t.field(PDF_SETTINGS, pdfSettings.toString()))
+			.transform((t) -> securitySettings == null ? t : t.field(SECURITY_SETTINGS, securitySettings.toString()))
+			.transform((t) -> {
+				try {
+					return settingsDoc == null ? t
+							: t.field(SETTING_DOC, settingsDoc.getInlineData(),
+									MediaType.MULTIPART_FORM_DATA_TYPE);
+				} catch (IOException e) {
+					// if we encounter an exception, then we just don't add this field. This should
+					// of error shouldn't ever happen.
+					return t;
+				}
```

Comment:
Can we do anything to clean up these lambdas?  This approach is not very readable.  Could we maybe move the getInlineData() calls out of the lambdas?

How about something like this:
```java
byte[] settingsData = settingsDoc != null ? settingsDoc.getInlineData() : null;
byte[] xmpDocData = xmpDoc != null ? xmpDoc.getInlineData() : null;
MultipartTransformer.create(multipart)
	.transform((t) -> fileTypeSettings == null ? t : t.field(FILE_TYPE_SETTINGS, fileTypeSettings))
	.transform((t) -> pdfSettings == null ? t : t.field(PDF_SETTINGS, pdfSettings.toString()))
	.transform((t) -> securitySettings == null ? t : t.field(SECURITY_SETTINGS, securitySettings.toString()))
	.transform((t) -> settingsData == null ? t : t.field(SETTING_DOC, settingsData, MediaType.MULTIPART_FORM_DATA_TYPE))
	.transform((t) -> xmpDocData == null ? t : t.field(XMP_DOC, xmpDocData, MediaType.MULTIPART_FORM_DATA_TYPE));
```
That looks a little more readable to me at the cost of a couple of temp variables.

Reply:
Done